app/code/Magento/Quote/Model/ResourceModel/Quote.php

```diff
@@ -316,4 +316,28 @@ public function save(\Magento\Framework\Model\AbstractModel $object)
 
         return $this;
     }
+
+    /**
+     * Quickly check if quote exists
+     *
+     * Uses direct DB query due to performance reasons
+     *
+     * @param int $quoteId
+     * @return bool
+     */
+    public function isQuoteExists(int $quoteId): bool
+    {
+        $connection = $this->getConnection();
+        $mainTable = $this->getMainTable();
+        $idFieldName = $this->getIdFieldName();
+
+        $field = $connection->quoteIdentifier(sprintf('%s.%s', $mainTable, $idFieldName));
+        $select = $connection->select()
+            ->from($mainTable, [$idFieldName])
+            ->where($field . '=?', $quoteId);
+
+        $data = $connection->fetchOne($select);
+
+        return (bool)$data;
```

Comment:
Why not simplifying like this?

```suggestion
        return (bool) $connection->fetchOne($select);
```

Reply:
I would say like this:
```suggestion
        return (bool)$connection->fetchOne($select);
```